server/routes/nextAppointment.ts

```diff
@@ -0,0 +1,288 @@
+      } else {
+        res.redirect(`/check-your-report/${req.params.id}`)
+      }
+    }
+  })
+
+  function validateNextAppointment(breachNotice: BreachNotice): ErrorMessages {
+    const errorMessages: ErrorMessages = {}
+    if (breachNotice.optionalNumberChecked) {
+      // Cannot be blank if set to use custom number
+      if (!breachNotice.optionalNumber || breachNotice.optionalNumber.trim().length === 0) {
+        errorMessages.contactByPhone = {
+          text: 'A number is required when using a different contact number.',
+        }
+      } else if (breachNotice.optionalNumber.length > 35) {
+        // Number needs to be 35 chars max to match NDelius DB
+        errorMessages.contactByPhone = {
+          text: 'The contact number entered is invalid. Numbers must be less than 35 characters long.',
+        }
+      } else {
+        const regex = /^[0-9]*$/
+        if (!regex.test(breachNotice.optionalNumber.replace(' ', ''))) {
+          errorMessages.contactByPhone = {
+            text: 'The contact number entered is invalid. Numbers must consist of numerals and spaces only.',
+          }
+        }
+      }
+    }
+    return errorMessages
+  }
+
+  async function showDraftPdf(id: string, res: Response) {
+    res.redirect(`/pdf/${id}`)
+  }
+
+  function initiateNextAppointmentRadioButtonsAndApplySavedSelections(
+    futureAppointments: Array<FutureAppointment>,
+    breachNotice: BreachNotice,
+  ): RadioButton[] {
+    // Read warning Types
+    const futureAppointmentRadioButtons: RadioButton[] = [
+      ...futureAppointments.map(futureAppointment => ({
+        text: [
+          futureAppointment.description,
+          futureAppointment.type.description,
+          DateTimeFormatter.ofPattern('d/M/yyyy').format(
+            DateTimeFormatter.ISO_LOCAL_DATE_TIME.parse(futureAppointment.datetime),
+          ),
+          DateTimeFormatter.ofPattern('HH:mm').format(
+            DateTimeFormatter.ISO_LOCAL_DATE_TIME.parse(futureAppointment.datetime),
+          ),
+          futureAppointment.location.buildingNumber.concat(` ${futureAppointment.location.streetName}`).trim(),
+          officerDisplayValue(futureAppointment.officer.name),
+        ]
+          .filter(item => item)
+          .join(', '),
+        value: `${futureAppointment.contactId}`,
+        selected: false,
+        checked: false,
+      })),
+    ]
+
+    // return warningTypeRadioButtons
+    // find currently selected code in breach notice and apply to the radio buttons
+    if (breachNotice.nextAppointmentId) {
+      futureAppointmentRadioButtons.forEach((radioButton: RadioButton) => {
+        if (breachNotice.nextAppointmentId && breachNotice.nextAppointmentId === Number(radioButton.value)) {
+          // eslint-disable-next-line no-param-reassign
+          radioButton.checked = true
+        }
+      })
+    }
+
+    return futureAppointmentRadioButtons
```

Comment:
No need for two loops here - you can set the checked flag directly in the map call.  Also you can use the common utils methods for date formatting.

```suggestion
    return futureAppointments.map(futureAppointment => ({
        text: [
          futureAppointment.description,
          futureAppointment.type.description,
          toUserDate(futureAppointment.datetime),
          toUserTime(futureAppointment.datetime),
         `${futureAppointment.location.buildingNumber} ${futureAppointment.location.streetName}`).trim(),
          officerDisplayValue(futureAppointment.officer.name),
        ].filter(item => item).join(', '),
        value: `${futureAppointment.contactId}`,
        selected: false,
        checked: breachNotice.nextAppointmentId && breachNotice.nextAppointmentId === futureAppointment.contactId,
      })),
    ]
```